R/get_national_data.R

```diff
@@ -0,0 +1,155 @@
+            } else {
+          stop("The country name is not recognised or returns more than one country. Suggest checking against UN standard spelling.")
+            }
+          }
+        }
+      }
+  }
+  
+  # Find the correct data-getter and ISO codes ----------------------------------------
+
+    get_data_function <- switch(source,
+                                "ecdc" = get_ecdc_cases,
+                                "who" = get_who_cases,
+                                stop("Unknown source entered"))
+    
+ 
+  # Get the data and country ISO codes ------------------------------------
+  data <- do.call(get_data_function, list())
+ 
+ # Filter to country -------------------------------------------------------
+ 
+ if (!is.null(input_country_name)){
+   
+   data <- data %>%
+     dplyr::mutate(country_lower = tolower(country)) %>%
+     dplyr::filter(country_lower %in% input_country_name) %>%
+     dplyr::select(-country_lower)
+   
+ }  
+ 
+ 
+  # Group data to country -----------------------------------
+    data <- data %>%
+      dplyr::group_by(country, iso_code)
+  
+  # Add columns that aren't there already, clean up data ------------------------------
+  data <- data %>%
+    dplyr::do(calculate_columns_from_existing_data(.)) %>%
+    add_extra_na_cols() %>%
+    set_negative_values_to_zero() %>%
+    dplyr::ungroup()
+  
+
+  # Totalise and return if totals data is requested -------------------------------------------
+  if (totals) {
+    
+    data <- data %>%
+      dplyr::rename(region_level_1 = country) %>%
+      totalise_data(include_level_2_regions = FALSE) %>%
+      dplyr::select(-recovered_total, -hosp_total, -tested_total) %>%
+      dplyr::arrange(-cases_total) %>%
+      dplyr::rename(country = region_level_1)
+    
+    return(tibble::tibble(data))
+  }
+
+  # Pad the data set ------------------------------------------------------------------
+  data <- data %>%
+    dplyr::rename("region_level_1" = "country") %>%
+    tidyr::drop_na(date) %>%
+    fill_empty_dates_with_na() %>%
+    complete_cumulative_columns() %>%
+    dplyr::rename("country" = "region_level_1")
+  
+  if(source == "ecdc"){
+    
+    data <- data %>%
+      dplyr::group_by(country) %>%
+      tidyr::fill(continent, population_2019, .direction = "up") %>%
+      dplyr::ungroup()
+    
+  }
+  
+  if(source == "who"){
+    
+    data <- data %>%
+      dplyr::group_by(country) %>%
+      tidyr::fill(who_region, .direction = "up") %>%
+      dplyr::ungroup()
+    
+  }
+  
+  # Arrange and return data -------------------------------------------------------
+  
+  data <- data %>%
+    dplyr::select(date, dplyr::everything(), country, iso_code, cases_new, cases_total, deaths_new, deaths_total,
+                  recovered_new, recovered_total, hosp_new, hosp_total,
+                  tested_new, tested_total) %>%
+    dplyr::arrange(date, country)
+  
+
+return(tibble::tibble(data))
```

Comment:
isn't it already definitely a tibble at this point? or no?

Reply:
Yes it is. I was keeping as much of Jonnie's code from the regional data function as I could for consistency, but not really necessary here. Removed